python/foglamp/common/configuration_manager.py

```diff
@@ -971,6 +971,65 @@ def diff(lst1, lst2):
 
         return result
 
+    async def delete_recursively_parent_category(self, category_name):
+        """Delete a parent-child relationship for a parent recursively 
```

Comment:
Delete a category and its child recursively including their parent-child relationship